### DISCUSSION TITLE
How does `fetcher` use `url` as an argument if it's never declared?

### DISCUSSION THREAD

**Discussion by platocrat** [vercel/swr#633] [General] [answered]
My GitHub issue got moved into discussion, so I thought this would be a better location for this since it relates to the `swr` library.

In the `index.js` file in this [`api-routes-rest` example](https://github.com/vercel/next.js/blob/canary/examples/api-routes-rest/pages/index.js) (shown below for convenience) I cannot understand why `fetcher` still under the `useSwr()` call:
<img width="504" alt="Screen Shot 2020-09-02 at 11 30 09 PM" src="https://user-images.githubusercontent.com/37757724/92071408-3f2b1a00-ed74-11ea-8607-8539e44620ae.png">

`fetch()` alone requires a valid url, an url which is not specified in the file, so how can line 7 in `pages/index.js`:
```javascript
const { data, error } = useSwr('/api/users', fetcher)
```
be returning assigning anything to data or error??

## Additional context

I am new to `useSwr` and maybe I am misunderstanding how the hook works, but here's what I see occuring:
1.  the `users` API endpoint is held contained under `/api/users.js` and may now be used inside other function components
2. `useSwr()` uses the `fetcher` function and the `users` API endpoint to return the json data and an error to be assigned to `data` and `error` respectively.

In this example with data made on the same origin, I assume that `useSwr()` just ignores the `fetcher` function and things still work because the data or error being returned is a result from the requests made to the `pages/api/users.js` endpoint, an endpoint which already has pre-made data fed in as input and returned as a response

However, with real API GET requests from a different origin, we require that the data is pulled using some data-fetching method. 

Assume we use native `fetch()` to create the following endpoint at `pages/api/realUsers.js`:
```javascript
const url = 'https://api.somejson'
const liveData = fetch(url).then(res => res.json())

export default async (req, res) => {
  res.json(liveData)
}
```

Now, how do I consume this endpoint by using the `useSwr('/api/liveData', fetcher)` hook?
I can no longer use the `fetcher` function as:
```javascript
const { data, error } = useSwr('/api/realUsers', fetcher)
```
because I already have my `liveData` contained within my liveData endpoint that I created using `fetch()`, and again, WHERE does the `fetcher` function obtain it's url from if I already used the appropriate url in creating my `liveData` endpoint??

**Comment by sergiodxa** [comment, marked as answer]
SWR falla your fetcher function passing the key as argument, in your case your URL is the key so your fetcher receives the URL from there.
